### PR TITLE
Fix get solidity version for udapp docs

### LINKED ISSUE #4834: improve getting solidity version
Here:
```
const getVersion = () => window.location.href.split('=')[5].split('+')[0].split('-')[1]
```
https://github.com/ethereum/remix-project/blob/master/libs/remix-ui/run-tab/src/lib/components/universalDappUI.tsx#L28

This getVersion is used to redirect users for receive ether link here:

https://github.com/ethereum/remix-project/blob/master/libs/remix-ui/run-tab/src/lib/components/universalDappUI.tsx#L361

- [ ] This logic should be improved as sometimes udapp starts loading before adding `version` field in URL params and it shows error in browser console with a blank screen on IDE halting IDE loading and not showing any error on screen. It should be caught and UI loading should not be stopped.

Screenshot 2024-06-03 at 7 29 49 PM

- [ ] Secondly, `receive` method was added in solidity v0.6.0 , may be we should check the version and redirect on relevant link

### PR DESCRIPTION
**Pull request by joeizang**
fixes #4834

### CODE REVIEW

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx [hunk 1]
```diff
@@ -32,6 +32,22 @@ export function ContractGUI(props: ContractGUIProps) {
   const initializeFields = useRef<Array<HTMLInputElement | null>>([])
   const basicInputRef = useRef<HTMLInputElement>()
   const intl = useIntl()
+  // const [solcVersion, setSolcVersion] = useState({ version: '', canReceive: true })
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:35]: remove commented code

File: libs/remix-ui/run-tab/src/lib/components/universalDappUI.tsx [hunk 2]
```diff
@@ -24,9 +24,22 @@ export function UniversalDappUI(props: UdappProps) {
   const [calldataValue, setCalldataValue] = useState<string>('')
   const [evmBC, setEvmBC] = useState(null)
   const [instanceBalance, setInstanceBalance] = useState(0)
-
-  const getVersion = () => window.location.href.split('=')[5].split('+')[0].split('-')[1]
-
+  // const [solcVersion, setSolcVersion] = useState({ version: '', canReceive: true })
+
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/components/universalDappUI.tsx:28]: remove commented code

File: libs/remix-ui/run-tab/src/lib/run-tab.tsx [hunk 3]
```diff
@@ -80,6 +80,22 @@ export function RunTabUI(props: RunTabProps) {
   const [runTab, dispatch] = useReducer(runTabReducer, initialState)
   const REACT_API = { runTab }
   const currentfile = plugin.config.get('currentFile')
+  const [solcVersion, setSolcVersion] = useState<{version: string, canReceive: boolean}>({ version: '', canReceive: true })
+
+  const getVersion = () => {
+    let version = ''
+    try {
+      version = window.location.href.split('=')[5].split('+')[0].split('-')[1].slice(1)
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/run-tab.tsx:88]: source of issue was this code. When there is no version in URL params, this throws. Better to check if version is available before splitting again

File: libs/remix-ui/run-tab/src/lib/components/universalDappUI.tsx (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/universalDappUI.tsx:28]: sure

File: libs/remix-ui/run-tab/src/lib/run-tab.tsx (hunk 3 above)
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/run-tab.tsx:88]: using null coalescing to make sure that there is a default compiler selected. Hope this is agreeable

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx (hunk 1 above)
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:35]: done

File: libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx [hunk 4]
```diff
@@ -299,8 +299,8 @@ export function ContractGUI(props: ContractGUIProps) {
           <div className="d-flex p-0 wrapperElement" onClick={handleActionClick} data-id={buttonOptions.dataId} data-title={buttonOptions.title}>
             <button
               className={`udapp_instanceButton text-nowrap overflow-hidden text-truncate ${props.widthClass} btn btn-sm ${buttonOptions.classList}`}
-              data-id={buttonOptions.dataId}
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:302]: Are these needed? `"firstbullseye"` and `"bullseye"`

Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx:302]: done

**Review by ioedeveloper — APPROVED** [review]

**Pull request merged**